modules/casper.js

```diff
@@ -2185,6 +2192,13 @@ Casper.prototype.waitForAlert = function(then, onTimeout, timeout) {
  */
 Casper.prototype.waitForPopup = function waitForPopup(urlPattern, then, onTimeout, timeout) {
     "use strict";
+    if (then && !utils.isFunction(then)) {
+        throw new CasperError("waitForPopup() next step definition must be a function");
+    }
+    timeout = ~~timeout;
+    if (timeout < 0) {
+        throw new CasperError("waitForPopup() only accepts an integer >= 0 as a timeout value");
+    }
     return this.waitFor(function() {
```

Comment:
Please put this block of code in a helper function and call it rather than copying it around.